python/tank/descriptor/bundle_cache_usage/manager.py

```diff
@@ -0,0 +1,270 @@
+                if not cls.__singleton_instance:
+                    cls.__singleton_instance = super(BundleCacheManager, cls).__new__(cls, *args, **kwargs)
+                    cls.__singleton_instance.__initialized = False
+
+        return cls.__singleton_instance
+
+    def __init__(self, bundle_cache_root):
+        super(BundleCacheManager, self).__init__()
+        log.debug_manager("__init__('%s')" % bundle_cache_root)
+        # TODO: returning would cause a silent non-usage of specified parameter
+        if (self.__initialized):
+            return
+        self._worker = None
+
+        if bundle_cache_root is None:
+            raise ValueError("The 'bundle_cache_root' parameter is None.")
+
+        if not os.path.exists(bundle_cache_root):
+            raise ValueError("The specified 'bundle_cache_root' parameter folder does not exists.")
+
+        if not os.path.isdir(bundle_cache_root):
+            raise ValueError("The specified 'bundle_cache_root' parameter is not a directory.")
+
+        self._bundle_cache_root = bundle_cache_root
+        self._worker = BundleCacheUsageWorker(bundle_cache_root)
+        self._worker.start()
+        self.__initialized = True
+
+    @classmethod
+    def delete_instance(cls):
+        with cls.__singleton_lock:
+            if cls.__singleton_instance:
+                if cls.__singleton_instance._worker:
+                    cls.__singleton_instance._worker.stop()
+                    cls.__singleton_instance = None
+
+    @classmethod
+    def _find_app_store_path(cls, base_folder):
+        for (dirpath, dirnames, filenames) in os.walk(base_folder):
+            if dirpath.endswith('app_store'):
+                return dirpath
+
+        return None
+
+    @classmethod
+    def _find_descriptors(cls, base_folder, max_walk_depth):
+        path_list = []
+        base_folder_len = len(base_folder)
+        for (dirpath, dirnames, filenames) in os.walk(base_folder):
+            shopped_base_path = dirpath[base_folder_len + 1:]
+            level_down_count = shopped_base_path.count(os.sep)
+            if level_down_count <= max_walk_depth:
+                for filename in filenames:
+                    if filename.endswith('info.yml'):
+                        path_list.append(dirpath)
+
+        return path_list
+
+    def _find_app_store_bundles(self):
+        # TODO: Find a tk-core reference about why MAX_DEPTH_WALK should be set to 2
+        MAX_DEPTH_WALK = 2
+
+        bundle_path_list = []
+        bundle_cache_root = self.bundle_cache_root
+
+        # Process the local app store first
+        bundle_cache_app_store = BundleCacheManager._find_app_store_path(bundle_cache_root)
+
+        if bundle_cache_app_store:
+            log.debug("Found local app store path: %s" % (bundle_cache_app_store))
+
+            if bundle_cache_app_store and \
+                    os.path.exists(bundle_cache_app_store) and \
+                    os.path.isdir(bundle_cache_app_store):
```

Comment:
`os.path.exists` is redundant. Not existing folders are not dirs

Reply:
No need to test again for the truthyness of `bundle_cache_app_store`, you already tested it above.